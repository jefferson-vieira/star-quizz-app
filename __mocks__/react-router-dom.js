import React from 'react';

const router = require('react-router-dom');

router.BrowserRouter = ({ children }) => <div>{children}</div>;

module.exports = router;
