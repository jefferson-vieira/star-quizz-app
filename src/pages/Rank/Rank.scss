@import 'assets/scss/variables';

.rank {
  h1 {
    color: $primary;
    font-weight: bold;
    font-size: 3.5rem;
  }

  &-list {
    &__message,
    &--text-white {
      &,
      &:hover {
        color: $white;
      }
    }
  }
}
