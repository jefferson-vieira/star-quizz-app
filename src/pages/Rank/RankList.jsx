import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faTrophy } from '@fortawesome/free-solid-svg-icons';

import Header from '@/components/Header';

import timeUtils from '@/utils/time'

library.add(faTrophy);

const renderRows = players => {
  return players.map((player, index) => (
    <tr key={index}>
      <th scope="row">
        {[1, 2, 3].includes(++index) ? (
          <FontAwesomeIcon
            icon="trophy"
            color={index === 1 ? 'gold' : index === 2 ? 'silver' : '#cd7f32'}
          />
        ) : (
          `${index}º`
        )}
      </th>
      <td>{player.score}</td>
      <td>{player.name}</td>
      <td>{player.date ? timeUtils.toLocalDateTime(player.date) : 'Desconhecido...'}</td>
      <td>
        <a className="rank-list--text-white" href={`mailto:${player.email}`}>
          {player.email}
        </a>
      </td>
    </tr>
  ));
};

const RankList = ({ players }) => (
  <section id="rank-list" className="container my-5">
    <Header title="Rank" />
    {players.length ? (
      <div className="table-responsive">
        <table className="table table-dark table-striped table-bordered table-hover text-center my-3">
          <caption className="sr-only">Rank</caption>
          <thead>
            <tr className="bg-primary">
              <th>#</th>
              <th>Pontuação</th>
              <th>Nome</th>
              <th>Data</th>
              <th>E-mail</th>
            </tr>
          </thead>
          <tbody>{renderRows(players)}</tbody>
        </table>
      </div>
    ) : (
      <div className="rank-list__message my-5">
        <h2>Ninguém jogou ainda.</h2>
        <h3>Seja o primeiro!</h3>
      </div>
    )}
  </section>
);

export default RankList;
