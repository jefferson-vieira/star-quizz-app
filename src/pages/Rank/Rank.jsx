import React, { Component } from 'react';

import PageHeader from '@/components/PageHeader';
import HomeButton from '@/components/HomeButton';

import RankList from './RankList';

class Rank extends Component {
  constructor(props) {
    super(props);

    this.state = {
      players: []
    };
  }

  componentDidMount() {
    const storage = JSON.parse(localStorage.getItem('scores')) || [];
    this.setState({ players: storage });
  }

  render() {
    const { players } = this.state;

    return (
      <section id="rank" className="rank container-fluid">
        <PageHeader />
        <RankList players={players} />
        <HomeButton />
      </section>
    );
  }
}

export default Rank;
