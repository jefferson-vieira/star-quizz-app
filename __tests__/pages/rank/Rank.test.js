import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import Rank from '../../../src/pages/Rank/Rank';
import RankList from '../../../src/pages/Rank/RankList';

describe('Testing Rank Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Rank />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should has a RankList', () => {
    const wrapper = mount(
      <MemoryRouter>
        <Rank />
      </MemoryRouter>
    );

    expect(wrapper.find(RankList).exists()).toBe(true);
  });

  it('should can redirect to /home', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/rank']}>
        <Rank />
      </MemoryRouter>
    );

    expect(wrapper.find('Link').props().to).toBe('/home');
  });
});
