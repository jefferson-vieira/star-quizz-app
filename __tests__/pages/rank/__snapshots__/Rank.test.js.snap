// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Rank Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rank />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageHeader />,
        <RankList
          players={Array []}
        />,
        <HomeButton />,
      ],
      "className": "rank container-fluid",
      "id": "rank",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "players": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageHeader />,
          <RankList
            players={Array []}
          />,
          <HomeButton />,
        ],
        "className": "rank container-fluid",
        "id": "rank",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "players": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
